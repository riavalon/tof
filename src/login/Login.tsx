import React, { useContext } from 'react'
import { useHistory } from 'react-router-dom'

import './Login.styles.scss'
import TextInput from '../components/form/textInput/TextInput'
import { AuthContext } from '../App'

const Login = () => {
  const context = useContext(AuthContext)
  const history = useHistory()
  const handleOnClick = function() {
    context.login()
    history.push('/admin')
  }
  return (
    <div className="page-container">
      <div className="tof-login">
        <div className="tof-login__form">
          <h2>Admin Login</h2>
          <TextInput className="tof-login__username" label="username" id="tof-test-login-username" handleChange={(val: any) => {}} />
          <TextInput className="tof-login__password" type="password" label="password" id="tof-test-login-password" handleChange={(val: any) => {}} />
          <button onClick={handleOnClick} className="tof-login__login-btn" data-testid="tof-test-login-button">
            Login
          </button>
        </div>
      </div>
    </div>
  )
}

export default Login