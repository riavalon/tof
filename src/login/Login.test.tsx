import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Login from './Login'

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn()
  })
}))

describe('Login', () => {
  
  it('should render a login form', () => {
    const { baseElement, queryByTestId, queryByText } = render(<Login />)

    expect(queryByText(/admin login/i)).toBeInTheDocument()
    expect(queryByTestId(/tof-test-login-username/i)).toBeInTheDocument()
    expect(queryByTestId(/tof-test-login-password/i)).toBeInTheDocument()
    expect(queryByTestId(/tof-test-login-button/i)).toBeInTheDocument()

    expect(baseElement).toMatchSnapshot()
  })
})