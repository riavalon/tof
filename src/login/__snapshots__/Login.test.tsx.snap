// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render a login form 1`] = `
<body>
  <div>
    <div
      class="page-container"
    >
      <div
        class="tof-login"
      >
        <div
          class="tof-login__form"
        >
          <h2>
            Admin Login
          </h2>
          <div
            class="tof-form-control tof-login__username"
          >
            <label
              for="tof-test-login-username"
            >
              username
            </label>
            <input
              data-testid="tof-test-login-username"
              id="tof-test-login-username"
              type="text"
            />
          </div>
          <div
            class="tof-form-control tof-login__password"
          >
            <label
              for="tof-test-login-password"
            >
              password
            </label>
            <input
              data-testid="tof-test-login-password"
              id="tof-test-login-password"
              type="password"
            />
          </div>
          <button
            class="tof-login__login-btn"
            data-testid="tof-test-login-button"
          >
            Login
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
